source/shared/cpp/ObjectModel/ActionParserRegistration.cpp

```diff
@@ -20,21 +20,25 @@ namespace AdaptiveSharedNamespace {
         });
     }
 
-    void ActionParserRegistration::AddParser(std::string const &elementType, std::shared_ptr<ActionElementParser> parser)
+    void ActionParserRegistration::AddParser(std::string const &elementType,
+        std::shared_ptr<ActionElementParser> parser)
     {
+        // make sure caller isn't attempting to overwrite a known element's parser
         if (m_knownElements.find(elementType) == m_knownElements.end())
         {
             ActionParserRegistration::m_cardElementParsers[elementType] = parser;
         }
         else
         {
-            throw AdaptiveCardParseException(ErrorStatusCode::UnsupportedParserOverride, "Overriding known action parsers is unsupported");
+            throw AdaptiveCardParseException(ErrorStatusCode::UnsupportedParserOverride,
+                "Overriding known action parsers is unsupported");
         }
     }
 
     void ActionParserRegistration::RemoveParser(std::string const &elementType)
     {
-        if (m_knownElements.find(elementType) != m_knownElements.end())
+        // make sure caller isn't attempting to remove a known element's parser
+        if (m_knownElements.find(elementType) == m_knownElements.end())
```

Comment:
I'm assuming that this was the intended behavior?

Reply:
Looks right. In ElementParserRegistration we throw in an else (as we do in AddParser). Maybe add that here as well?

---
In reply to: [208670589](https://github.com/Microsoft/AdaptiveCards/pull/1776#discussion_r208670589) [](ancestors = 208670589)

Reply:
Or alternatively don't throw in the element parser registration.

---
In reply to: [208674103](https://github.com/Microsoft/AdaptiveCards/pull/1776#discussion_r208674103) [](ancestors = 208674103,208670589)

Reply:
I think throwing makes sense. I'll add it in :)

Reply:
No only does it make sense, but it is consistent with how ElementParserRegistration works :)